src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -1,1293 +1,1316 @@
+            {
+                if (taskHub == null
+                    && key.Equals(TaskHubParameter, StringComparison.OrdinalIgnoreCase)
+                    && !string.IsNullOrWhiteSpace(pairs[key]))
+                {
+                    taskHub = pairs[key];
+                }
+                else if (connectionName == null
+                    && key.Equals(ConnectionParameter, StringComparison.OrdinalIgnoreCase)
+                    && !string.IsNullOrWhiteSpace(pairs[key]))
+                {
+                    connectionName = pairs[key];
+                }
+            }
+
+            var attribute = new DurableClientAttribute
+            {
+                TaskHub = taskHub,
+                ConnectionName = connectionName,
+            };
+
+            return this.GetClient(attribute);
+        }
+
+        // protected virtual to allow mocking in unit tests.
+        protected virtual IDurableClient GetClient(DurableClientAttribute attribute)
+        {
+            return this.config.GetClient(attribute);
+        }
+
+        internal string GetBaseUrl()
+        {
+            Uri notificationUri = this.GetWebhookUri();
+
+            string hostUrl = notificationUri.GetLeftPart(UriPartial.Authority);
+            return hostUrl + notificationUri.AbsolutePath.TrimEnd('/');
+        }
+
+        internal string GetUniversalQueryStrings()
+        {
+            Uri notificationUri = this.GetWebhookUri();
+
+            return !string.IsNullOrEmpty(notificationUri.Query)
+                ? notificationUri.Query.TrimStart('?')
+                : string.Empty;
+        }
+
+        internal HttpCreationPayload GetInstanceCreationLinks()
+        {
+            string baseUrl = this.GetBaseUrl();
+            string instancePrefix = baseUrl + "/" + OrchestratorsControllerSegment + "{functionName}[/{instanceId}]";
+
+            string querySuffix = this.GetUniversalQueryStrings();
+
+            HttpCreationPayload httpCreationPayload = new HttpCreationPayload
+            {
+                CreateNewInstancePostUri = instancePrefix + "?" + querySuffix,
+                CreateAndWaitOnNewInstancePostUri = instancePrefix + "?timeout={timeoutInSeconds}&pollingInterval={intervalInSeconds}&" + querySuffix,
+            };
+
+            return httpCreationPayload;
+        }
+
+        private HttpManagementPayload GetClientResponseLinks(
+             HttpRequestMessage request,
+             string instanceId,
+             string taskHubName,
+             string connectionName,
+             bool returnInternalServerErrorOnFailure = false)
+        {
+            Uri notificationUri = this.GetWebhookUri();
+            Uri baseUri = request?.RequestUri ?? notificationUri;
+
+            // Check if the environment variable is set
+            bool useForwardedHost = Environment.GetEnvironmentVariable("DurableFunctions__UseForwardedHost")?.Equals("true", StringComparison.OrdinalIgnoreCase) == true;
```

Comment:
It would be better if we could use the new options you added to `HttpOptions.cs` rather than an environment variable here. That way users can configure the setting through host.json (which also supports environment variable overrides). I think you can simply change this current line to the following:

```suggestion
            bool useForwardedHost = this.config.Options.HttpOptions.UseForwardedHost;
```